added termonad, some deps, and also hs-duktape

formula-spectre opened the pull request:
hope I did this right. I added termonad, focuslist, gi-vte, xml-html-qq; then I also added hs-duktape.

solpeth commented:
Thanks for the PR! If you've generated everything with hackport, most things are usually correct.

It seems `pkgcheck` CI is failing on your `-9999` version (live ebuild) of termonad, which can be fixed either by hiding the keywords from it or deleting the live ebuild altogether. Given you have also provided a version of termonad, is the live ebuild really needed? Unless it is really necessary, they tend to not get looked after over time.

Thanks again. Let me know if I can help.

formula-spectre commented:
removed termonad-9999, I added it because I thought that people might want the latest git version. but no problem, this should do it, let me know if now it's fine

hololeap commented:
Please format your commits so that they look like: `dev-haskell/focuslist: Add 0.1.0.2`, where each commit is only modifying one package. This will make it easier to keep track of changes and revert if necessary. Using `repoman commit` to commit each package will help with this and will help you spot QA issues sooner.

hololeap commented:
`focuslist` has a missing dependency: `>=dev-haskell/genvalidity-containers-0.5`

formula-spectre commented:
- done! did not know about that. hope I did it right this time
- does it? I see it at line 30 of the ebuild? are you referring to a genereal dep outside of ```test```?
- done! think that is what you meant
- uh.. I have no idea on the last one sorry.. saying "it builds on me" is dumb because i do acknowledge that it's not that reproducible.

hololeap commented:
* This is better, although 4e9735206295fe878151fd454caa04a2676a9a27 seems to be doing quite a bit more than the commit message suggests. 83c351fb25f2ca849a690ea487279e1690aff7f7 is also misspelled.
* The problem here is that `dev-haskell/genvalidity-containers` does not exist in the tree, so you will need to run `hackport merge genvalidity-containers`, then check for any missing dependencies, test and commit
* This doesn't seem to have been changed: https://github.com/gentoo-haskell/gentoo-haskell/pull/1232/commits/83c351fb25f2ca849a690ea487279e1690aff7f7#diff-fcf116d3ad63e36564adf32c8936b59d9bbe0cb23e544e57046b407321556af9R18
* You would need to install with `FEATURES=test` and `USE=test` to hit the error I am talking about.

I am fixing this branch for you and can pull it in when I'm done, but this should give you an idea of what to do for future PRs.

hololeap commented:
I pulled the edited version of your branch. You can use [my `termonad` branch](https://github.com/hololeap/gentoo-haskell/pull/1) for reference for future pull requests.

formula-spectre commented:
well that was  a bit embarassing from my side, but I will take this as a reference for future pull requests. thank you for your patience and for accepting (although not my branch) the packages!

solpeth commented:
@formula-spectre no embarrassment whatsoever! Thanks very much for the PR and I look forward to more in the future. Every little bit helps. We're here to help, so do feel free to reach out if you ever need a hand.

hololeap commented:
Yeah, there is definitely no need to feel embarassed. I decided that completely reworking your PR was the fastest way to get the software you wanted into the tree and to show you the best practices I've learned over the couple years of working with this repo. Don't hesitate to submit more PRs; just try to use this one as a reference.

hololeap commented:
I'll work on cataloging these best practices in [the wiki](https://github.com/gentoo-haskell/gentoo-haskell/wiki)
